### DISCUSSION TITLE
Not working another build ..

### DISCUSSION THREAD

**Discussion by sorydima** [codemagic-ci-cd/codemagic-docs#2559] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Run the https://github.com/sorydima/Katya-.git

on CodeMagic for the iOS & Android!

### Expected results

Builds done ..))

### Actual results

![Снимок экрана от 2024-05-27 15-12-16](https://github.com/codemagic-ci-cd/codemagic-docs/assets/43252008/401eecf0-104e-45ef-997a-6b09c1311f79)

![Снимок экрана от 2024-05-27 15-12-34](https://github.com/codemagic-ci-cd/codemagic-docs/assets/43252008/4bcb5472-5e56-4cb8-888b-0901cfafb5bf)

Issues such as these may happen due to a number of reason. Please verify that all the dependencies you are using locally are also available to Codemagic. Here's what you can try:

If you have recently moved your Flutter project to another location, try rescanning the application in App settings > Repository settings > Rescan application
If pubspec.yaml is not found in an unpublished package, please check that specified local dependency exists and is a flutter/dart dependency. If the failed dependency is a private submodule from another account, please make sure to provide Codemagic access to it to be able to clone. More information on it can be found at [Accessing private repositories](https://docs.codemagic.io/knowledge-git/access-private-git-submodules/?utm_source=app&utm_campaign=git-submodules)
If your dependency is in a public repository and you want to get anonymous access to it (without using a private key), please use https link (not ssh).
In case you depend on a specific commit, branch, or tag (ref argument is added), please ensure you provided the ssh url for the git dependency (not https url)

Build overview
ID:
665477fd574cbb9ede4de32f
Please provide this ID when you contact our support team
Index:
5
Status:
failed
Workflow:
Default Workflow
Started by:
Dmitry Sorokin (@sorydima)
Dmitry Sorokin (@sorydima)sorydima@gmail.com
Started:
2 minutes ago
May 27th, 2024 at 15:09 MSK
Duration:
1m 40s
Machine:
Mac mini M1
Branch:
main
Commit:
[3390bb0](https://github.com/sorydima/Katya-/commit/3390bb02204cd5e95b091595191ea346b8d69c60)
Current configuration
Flutter channel:
master
Mode:
release
Xcode version:
edge
Build for:
iOS & Android

### Build id (optional)

665477fd574cbb9ede4de32f

**Comment by icarusdust** [comment]
Hi @sorydima, please note that each step in the build has its logs printed out to make it easier for debugging. If you click on the Installing dependencies step which is the failing one, you should notice the reason why your build is failing. 

It is usually about a dependency version that is required to be updated because of your project requirements.  And as you are building a Flutter application, then you can update the highlighted dependency version in `pubspec.yaml` under `dependencies`.  Let me know if it helps you.

**Reply by sorydima** [reply]
Could you please help me with the project to make it done?

**Reply by icarusdust** [reply]
Did you click to open the `Installing dependencies` step ? If so, let me know what dependency it is. Otherwise, you can find the dependency in `pubspec.yaml` under `dependencies` and update its version to the required version, and the required version is already highlighted in the error logs:
```
dependencies:
  flutter:
    sdk: flutter
  flutter_localizations:
    sdk: flutter

  ### state
  equatable: ^2.0.0 # e.g. here is where you need to update the version for the problematic dependency
```

**Reply by sorydima** [reply]
Sorry, but I do not really understand what dependencies should I upgrade? Could you check, please 🙏

**Reply by icarusdust** [reply]
You need to click on the red line to see which dependency it is. So, go to your failing build, click on the red line (Installing dependencies step), and the logs will be visible. It would be better if you could check it yourself, so you can understand what the issue is. When you see the logs, then you will see which dependency needs to be updated. Let me know if you are still stuck after it.

**Comment by icarusdust** [comment, marked as answer]
I am closing the thread due to inactivity @sorydima You can reply to this thread if you have any questions.
